federation/server.c

```diff
@@ -0,0 +1,767 @@
+#include "c-spiffe/federation/server.h"
+#include "c-spiffe/logger/logger.h"
+#include "c-spiffe/spiffetls/spiffetls.h"
+#include "c-spiffe/utils/error.h"
+#include "c-spiffe/utils/picohttpparser.h"
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+
+spiffebundle_EndpointInfo *spiffebundle_EndpointInfo_New()
+{
+    spiffebundle_EndpointInfo *e_info = calloc(1, sizeof(*e_info));
+    mtx_init(&e_info->mutex, mtx_plain);
+    return e_info;
+}
+
+err_t spiffebundle_EndpointInfo_Free(spiffebundle_EndpointInfo *e_info)
+{
+    // logger_FmtPush(LOGGER_DEBUG, "spiffebundle_EndpointInfo_Free %p",
+    // e_info);
+    if(!e_info) {
```

Comment:
I think you should ignore when the pointer is NULL rather than returning error

Reply:
The user can always ignore the return value, and returning the error allows for "after the fact" checking, in line with most of the library